tsec-libsodium/src/main/scala/tsec/libsodium/kx/KeyExchange.scala

```diff
@@ -0,0 +1,54 @@
+package tsec.libsodium.kx
+
+import cats.effect.Sync
+import tsec.libsodium.ScalaSodium
+
+object KeyExchange {
+
+  def generateKeyPair[F[_]](implicit F: Sync[F], S: ScalaSodium): F[SodiumKeyPair] = F.delay {
+    val pk = new Array[Byte](ScalaSodium.crypto_kx_PUBLICKEYBYTES)
+    val sk = new Array[Byte](ScalaSodium.crypto_kx_SECRETKEYBYTES)
+
+    S.crypto_kx_keypair(pk, sk)
+
+    SodiumKeyPair(SodiumKey$$.is[PublicKey].coerce(pk), SodiumKey$$.is[SecretKey].coerce(sk))
+  }
+
+  def generateKeyPairSeed[F[_]](seed: Array[Byte])(implicit F: Sync[F], S: ScalaSodium): F[SodiumKeyPair] = F.delay {
+    if (seed.length != ScalaSodium.crypto_kx_SEEDBYTES)
+      throw KeySeedingError(seed.length)
+
+    val pk = new Array[Byte](ScalaSodium.crypto_kx_PUBLICKEYBYTES)
+    val sk = new Array[Byte](ScalaSodium.crypto_kx_SECRETKEYBYTES)
+
+    S.crypto_kx_seed_keypair(pk, sk, seed)
+
+    SodiumKeyPair(SodiumKey$$.is[PublicKey].coerce(pk), SodiumKey$$.is[SecretKey].coerce(sk))
```

Comment:
We should probably try not to reference the newtype outside of the file. `SodiumKey[A]` does the same thing